cylc/flow/scripts/lint.py

```diff
@@ -149,7 +149,7 @@
         'url': 'https://github.com/cylc/cylc-flow/issues/3825',
         'index': 10
     },
-    re.compile(r'#.*?{[{%]'): {
+    re.compile(r'(?<!{)#.*?{[{%]'): {
         'short': 'Cylc will process commented Jinja2!',
```

Comment:
A small point (and not your change @ColemanTom - but since we're here). This warning is arguably technically inaccurate (Jinja2 will process the Jinja2 code);  and ambiguous (users might think it applies to Jinja2 comments too.

Maybe change to this?

```suggestion
        'short': 'Cylc comments (#) do not prevent Jinja2 processing.',
```

(Requires agreement from @wxtim and @oliver-sanders)

Reply:
I think that's an improvement on what I originally wrote. If @oliver-sanders is happy - pls merge.